staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -144,9 +145,48 @@ func (t *prefixTransformers) TransformFromStorage(data []byte, context Context)
 			} else {
 				RecordTransformation("from_storage", string(transformer.Prefix), start, err)
 			}
+
+			// It is valid to have overlapping prefixes when the same encryption provider
+			// is specified multiple times but with different keys (the first provider is
+			// being rotated to and some later provider is being rotated away from).
+			//
+			// Example:
+			//
+			//  {
+			//    "aescbc": {
+			//      "keys": [
+			//        {
+			//          "name": "2",
+			//          "secret": "some key 2"
+			//        }
+			//      ]
+			//    }
+			//  },
+			//  {
+			//    "aescbc": {
+			//      "keys": [
+			//        {
+			//          "name": "1",
+			//          "secret": "some key 1"
+			//        }
+			//      ]
+			//    }
+			//  },
+			//
+			// The transformers for both aescbc configs share the prefix k8s:enc:aescbc:v1:
+			// but a failure in the first one should not prevent a later match from being attempted.
+			// Thus we never short-circuit on a prefix match that results in an error.
+			if err != nil {
+				lastErr = err
```

Comment:
Should we use the last err or the first err?

Reply:
I do not know if there a good way to say which error is the most useful.  They are all equal in my mind because they match the required prefix.

Reply:
Also they are likely to all be the same, some variation on:

```
no matching key was found for the provided AES transformer
```

Reply:
> Also they are likely to all be the same, some variation on:
> 
> ```
> no matching key was found for the provided AES transformer
> ```

Hmm that may be too broad in the case of you accidentally changing the key but not its name.  I can aggregate them all?

Reply:
Or summarize?  If we have to live debug this it’s has gotten really real and we’re likely to be panicking

Reply:
Fixed.